### ISSUE TITLE
[Package]: ksh93u+m

### ISSUE THREAD

**Issue by numberokta** [termux/termux-packages#19824]
### Why is it worth to add this package?

This is very good version of `AT&T` based korn shell that now, support and develop by another maintainer. It's a very fast and full-featured shell and have so many builtin features that other shells don't have. Also, it's different from other korn shell like `mksh`, `lokesh`.

### Home page URL

http://www.kornshell.com/

### Source code URL

https://github.com/ksh93/ksh

### Packaging policy acknowledgement

- [X] I certify that I have read [Termux Packaging Policy](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy) and understand that my request will be denied in case of violation.

**Comment by TomJo2000** [comment]
Seeing as `bash` is a nearly perfect superset of `ksh` I am inclined to veto packaging `ksh` on "duplicated functionality" grounds.

However, I do realize it is a historically relevant Shell, and having it available may be beneficial.
Though general compatibility with existing Ksh scripts, utilizing Ksh specific syntax or quirks seems rather dubious, as obviously Termux has not been a viable target for such.

Overall, I think packaging Ksh would be of very dubious value beyond the vanity of having it available.

**Comment by truboxl** [comment]
Duplicate of #1341

**Comment by numberokta** [comment]
I tested many shell codes in bash and ksh93, and it had better performance in ksh93, and so many commands in ksh93 are builtin that also make it faster. You can see all features of bash and ksh93 [here](https://docstore.mik.ua/orelly/unix3/unixnut/ch04_09.htm)

**Comment by TomJo2000** [comment]
This is still a duplicate of #1341, `loksh` is the Linux port of the OpenBSD version of Ksh, it should be functionally equivalen to Ksh93.

**Comment by numberokta** [comment]
@TomJo2000 It's not. I test that before.

**Comment by numberokta** [comment]
I tried to build it myself, but it gives error. any help will be great.

**Comment by TomJo2000** [comment]
You can request it in the TUR if it absolutely must be AT&T Ksh93.
(In which case, what the hell to you have specifically requiring that version?)
https://github.com/termux-user-repository/tur

As for the main repo, it would be a nearly identical copy of `loksh`.

**Comment by numberokta** [comment]
Yes, both are korn shell And they are similar in commands and appearance but different in source code and performance and builtin commands.
Anyway, thanks. I will request in TUR

**Comment by numberokta** [comment]
Also, some functions are missing, like arithmetic for loop, some array manipulation and somethings like that. So they are not identical. I tested both mksh and loksh, and even they are not identical.
